client/main.js

```diff
@@ -85,55 +83,21 @@ Array.from( document.getElementsByClassName( 'wcc-root' ) ).forEach( ( container
 	const store = compose( ...enhancers )( createStore )( Route.getReducer(), initialState );
 
 	if ( Route.getInitialActions ) {
-		_.forEach( Route.getInitialActions(), store.dispatch );
+		Route.getInitialActions().forEach( store.dispatch );
 	}
 
-	window.addEventListener( 'beforeunload', ( event ) => {
+	window.addEventListener( 'beforeunload', () => {
 		const state = store.getState();
 
 		if ( window.persistState ) {
 			storageUtils.setWithExpiry( persistedStateKey, Route.getStateForPersisting( state ) );
-			return;
 		}
-
-		//this only handles the shipping service settings page. Other pages use ProtectFormGuard and Component state
-		//TODO: remove once the service settings are Calypso-fied
```

Comment:
This happen like a year ago :)

I can confirm that the "You have unsaved changes" prompt works in the USPS service settings page, even without this code.